src/dict.c

```diff
@@ -1423,6 +1428,23 @@ static int _dictExpandIfNeeded(dict *d)
     return DICT_OK;
 }
 
+static int _dictShrinkIfNeeded(dict *d) {
+    /* Incremental rehashing already in progress. Return. */
+    if (dictIsRehashing(d)) return DICT_OK;
+    
+    if (DICTHT_SIZE(d->ht_size_exp[0]) > DICT_HT_INITIAL_SIZE && 
+        ((dict_can_resize == DICT_RESIZE_ENABLE &&  
+          d->ht_used[0] * 100 / DICTHT_SIZE(d->ht_size_exp[0]) < HASHTABLE_MIN_FILL) ||
+         (dict_can_resize != DICT_RESIZE_FORBID &&
+          d->ht_used[0] * 100 / DICTHT_SIZE(d->ht_size_exp[0]) < dict_force_shrink_ratio))) 
+    {
+        if (!dictTypeExpandAllowed(d))
```

Comment:
Would it be more appropriate to name this function "dictTypeReSizeAllowed" or "dictTypeReHashAllowed"? After all, the words "expand" and "shrink" have opposite meanings.

Currently, in the dict structure, the term "expand" is used to represent both rehashing and resizing operations. Even when the dictionary is shrinking, the same function is used. I believe it is necessary to review and modify the function names in a unified manner.

Reply:
I think "rehash" is one word, so dictTypeRehashAllowed is better.

Note that when we're shrinking, we don't need to compute the hash, so it's not really rehashing. Maybe "resize allowed" is better then?

Reply:
"resize" seems fine to me, i'm also ok with "rehash".
if we feel bad with "allowed" we can rename to "advised".

Reply:
OK. I have changed it to "resize".